internal/routing/reader.go

```diff
@@ -88,6 +88,56 @@ func (r *routing) LocalSubnet() (defaultSubnet net.IPNet, err error) {
 	return defaultSubnet, fmt.Errorf("cannot find default subnet in %d routes", len(routes))
 }
 
+func (r *routing) LocalSubnets() (localSubnets []net.IPNet, err error) {
+	links, err := netlink.LinkList()
+	if err != nil {
+		return localSubnets, fmt.Errorf("cannot find local subnet: %w", err)
+	}
+
+	localLinks := make(map[int]bool)
+
+	for _, link := range links {
+		if link.Attrs().EncapType != "ether" {
+			continue
+		}
+
+		localLinks[link.Attrs().Index] = true
+		if r.verbose {
+			r.logger.Info("local ethernet link found: %s", link.Attrs().Name)
+		}
+	}
+
+	if len(localLinks) == 0 {
+		return localSubnets, fmt.Errorf("cannot find any local interfaces")
+	}
+
+	routes, err := netlink.RouteList(nil, netlink.FAMILY_ALL)
+	if err != nil {
+		return localSubnets, fmt.Errorf("cannot list local routes: %w", err)
+	}
+
+	localSubnets = make([]net.IPNet, 1)
+
+	for _, route := range routes {
+		if route.Gw != nil || !localLinks[route.LinkIndex] {
+			continue
+		}
```

Comment:
Adapting to that set-like map mentioned above + adding nil check for `route.Dst`

```suggestion
		if route.Gw != nil || route.Dst == nil {
			continue
		} else if _, ok := localLinks[route.LinkIndex]; !ok {
			continue
		}
```